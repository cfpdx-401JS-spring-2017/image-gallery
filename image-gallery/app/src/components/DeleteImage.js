import React from 'react';
import PropTypes from 'prop-types';

export default function DeleteImage({ removeImage }) {

DeleteImage.propTypes = {
   removeImage: PropTypes.func.isRequired
}


return (
    <button className="delete-button"
        onClick={removeImage}>
        DELETE IMAGE
            </button>
)

}