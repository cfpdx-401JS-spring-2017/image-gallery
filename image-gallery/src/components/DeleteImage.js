import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class DeleteImage extends Component {

    static propTypes = {
        onDelete: PropTypes.func.isRequired
    }

    render() {
        return (
            <button className="delete-button"
                onClick={this.props.onDelete}>
                DELETE IMAGE
            </button>
        )
    }

}