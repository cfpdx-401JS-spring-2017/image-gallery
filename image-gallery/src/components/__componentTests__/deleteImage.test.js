import React from 'react';
import renderer from 'react-test-renderer';
import DeleteImage from '../../components/DeleteImage';

function handleDelete() {}

describe('DeleteImage', () => {
  it('deletes an images', () => {
    const tree = renderer.create(<DeleteImage onDelete={handleDelete} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});