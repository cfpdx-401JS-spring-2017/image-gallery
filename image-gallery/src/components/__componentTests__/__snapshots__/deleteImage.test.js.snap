// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteImage deletes an images 1`] = `
<button
  className="delete-button"
  onClick={[Function]}
>
  DELETE IMAGE
</button>
`;
